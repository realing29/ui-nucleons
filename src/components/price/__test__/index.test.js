import React from 'react';
import Price from '../';
import { shallow } from 'enzyme';

let price;

describe('<Price />', () => {
  let props = {
    value: 100.578,
    currencySign: 'RUB',
    beforePrice: true,
    withFractionalPart: true,
  };

  it('renders correctly', () => {
    price = shallow(<Price {...props} />);
    expect(price).toMatchSnapshot();
    props = { value: 100, currencySign: 'USD' };
    price = shallow(<Price {...props} />);
    expect(price).toMatchSnapshot();
    props = { value: 100, currencySign: 'USD', withFractionalPart: true, boldIntegerPart: true, fractionalAtTop: true };
    price = shallow(<Price {...props} />);
    expect(price).toMatchSnapshot();
    props = { value: 100, className: 'test', currencySign: 'RUB' };
    price = shallow(<Price {...props} />);
    expect(price.prop('className')).toEqual('price test');
  });

  it('has class "sign" only when current currency is rubles', () => {
    props = {
      value: 100.578,
      currencySign: 'RUB',
      beforePrice: true,
      withFractionalPart: true,
    };
    price = shallow(<Price {...props} />);
    expect(price.find('span').first().children().first().hasClass('sign')).toBeTruthy();
    price.setProps({ currencySign: 'USD' });
    expect(price.find('span').first().children().first().hasClass('sign')).toBeFalsy();
  });
});
