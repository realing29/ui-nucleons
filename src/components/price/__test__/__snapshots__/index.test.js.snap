// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price /> renders correctly 1`] = `
<span
  className="price"
>
  <span
    className="sign"
  >
    RUB 
  </span>
  <span>
    100
    .
    58
  </span>
</span>
`;

exports[`<Price /> renders correctly 2`] = `
<span
  className="price"
>
  <span>
    100
  </span>
  <span
    className={null}
  >
     USD
  </span>
</span>
`;

exports[`<Price /> renders correctly 3`] = `
<span
  className="price"
>
  <span>
    <b>
      100
    </b>
    .
    00
  </span>
  <span
    className={null}
  >
     USD
  </span>
</span>
`;
